api/valor_api/crud/_create.py

```diff
@@ -107,27 +107,31 @@ def create_or_get_evaluations(
     tuple[list[schemas.EvaluatationResponse], list[schemas.EvaluatationResponse]]
         Tuple of evaluation id lists following the form ([created], [existing])
     """
-    created, existing = backend.create_or_get_evaluations(db, job_request)
-
-    # start computations
-    for evaluation in created:
-        match evaluation.parameters.task_type:
-            case enums.TaskType.CLASSIFICATION:
-                compute_func = backend.compute_clf_metrics
-            case enums.TaskType.OBJECT_DETECTION:
-                compute_func = backend.compute_detection_metrics
-            case enums.TaskType.SEMANTIC_SEGMENTATION:
-                compute_func = backend.compute_semantic_segmentation_metrics
-            case _:
-                raise RuntimeError
-
-        if task_handler:
-            task_handler.add_task(
-                compute_func,
-                db=db,
-                evaluation_id=evaluation.id,
-            )
-        else:
-            compute_func(db=db, evaluation_id=evaluation.id)
-
-    return created + existing
+    evaluations = backend.create_or_get_evaluations(db, job_request)
+    for evaluation in evaluations:
+        if evaluation.status in {
+            enums.EvaluationStatus.PENDING,
+            enums.EvaluationStatus.FAILED,
+        }:
+            match evaluation.parameters.task_type:
+                case enums.TaskType.CLASSIFICATION:
+                    compute_func = backend.compute_clf_metrics
+                case enums.TaskType.OBJECT_DETECTION:
+                    compute_func = backend.compute_detection_metrics
+                case enums.TaskType.SEMANTIC_SEGMENTATION:
+                    compute_func = (
+                        backend.compute_semantic_segmentation_metrics
+                    )
+                case _:
+                    raise RuntimeError
+
+            if task_handler:
+                task_handler.add_task(
+                    compute_func,
+                    db=db,
+                    evaluation_id=evaluation.id,
+                )
```

Comment:
Would you prefer a `retry_evaluation` endpoint? This would allow completed evaluations to be re-run too which may be nice in dev instead of wiping the db.

Reply:
perhaps we should require the user to explicitely state that they want to retry failed evaluations (e.g., by passing an `allow_retries=True` argument to the evaluation call)?

Reply:
Im open to either, @rsbowman-striveworks  what are your thoughts?

Reply:
I think either of these works.  Maybe this *is* a good place to put this, with Nick's suggestion of having a flag controlling whether to retry failed evalutions.